library/farmosUtil/farmosUtil.soilDisturbanceTerminationLog.unit.cy.js

```diff
@@ -0,0 +1,630 @@
+        statusCode: 401,
+      });
+
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedPlantAsset) => {
+            cy.wrap(
+              farmosUtil
+                .createSoilDisturbanceTerminationLog(
+                  '2023-11-20',
+                  'CHUAU', // Location
+                  [], // No beds to terminate
+                  updatedPlantAsset // Pass the plant asset with no beds
+                )
+                .then(() => {
+                  throw new Error(
+                    'Creating soil disturbance log for plant with no beds should have failed.'
+                  );
+                })
+                .catch((error) => {
+                  expect(error.message).to.equal(
+                    'Request failed with status code 401'
+                  );
+                })
+            );
+          }
+        );
+      });
+
+      // Recheck the plant asset to ensure its state remains intact
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedAsset) => {
+            expect(updatedAsset.relationships.location).to.have.length(0); // No beds should exist
+            expect(updatedAsset.attributes.status).to.not.equal('archived'); // Asset should not be archived
+          }
+        );
+      });
+
+      // Delete the plant asset
+      cleanupLogsAndAsset({
+        deletePlantAsset: true,
+      });
+    }
+  );
+
+  it('Delete a soil disturbance termination log', () => {
+    createPlantAsset(
+      'Test Plant Asset for Deletion',
+      'HERB-CILANTRO',
+      '2023-11-20'
+    );
+
+    cy.get('@newPlantAsset').then((plantAsset) => {
+      createMovementLog(
+        plantAsset,
+        ['ALF', 'ALF-1', 'ALF-3'],
+        ['seeding', 'tillage'],
+        '2023-11-20'
+      );
+    });
+
+    // Create the soil disturbance termination log for a specific bed
+    cy.get('@newPlantAsset').then((plantAsset) => {
+      cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+        (updatedPlantAsset) => {
+          cy.wrap(
+            farmosUtil.createSoilDisturbanceTerminationLog(
+              '2023-11-20',
+              'ALF',
+              ['ALF-3'], // Terminate ALF-3
+              updatedPlantAsset // Use the updated plant asset
+            )
+          ).as('soilDisturbanceLog');
+        }
+      );
+    });
+
+    // Delete the soil disturbance termination log
+    cy.get('@soilDisturbanceLog').then((soilDisturbanceLog) => {
+      cy.wrap(
+        farmosUtil.deleteSoilDisturbanceTerminationLog(soilDisturbanceLog.id)
+      ).then((result) => {
+        expect(result.status).to.equal(204); // Successful deletion
+      });
+    });
+
+    // Delete the movement log and plant asset
+    cleanupLogsAndAsset({
+      deleteMovementLog: true,
+      deletePlantAsset: true,
+    });
+  });
+
```

Comment:
Should we have a test for an error getting a termination log?

Reply:
The other log tests in `farmosUtil` included checks for errors when deleting logs, so I figured it made sense to follow the same testing structure here for consistency.